xgi/generators/uniform.py

```diff
@@ -346,13 +460,23 @@ def _index_to_edge(index, n, m):
     return [(index // (n**r) % n) for r in range(m - 1, -1, -1)]
 
 
-def _index_to_edge_partition(index, partition_sizes, m):
-    """Generate a hyperedge given an index in the list of possible edges
-    and a partition of community labels.
+def _index_to_edge_comb(index, n, m):
+    """Generate a hyperedge from an index given the number of nodes and size of hyperedges.
+
+    Imagine that there is a hypergraph with 4 nodes and an edge size of 3.
+    We write out each edge incrementing the last entry first, followed by the
+    second-to-last entry and so on, with each edge corresponding to an index
+    starting at zero. For example, (0, 1, 2) has index 0, (0, 1, 3) has index 0,
+    (0, 2, 3) has index 2, and (1, 2, 3) has index 3. This function will, for instance,
+    return (0, 2, 3) for index 2, network size 4, and edge size 3.
+
+
+    In this function, we prohibit multiedges, so each index corresponds to a
+    unique edge.
```

Comment:
But in _index_to_edge_prod too right? Should it not be, "so each edge corresponds to a unique index"?
Which is not he case in _index_to_edge_prod, eg index 1 and 4 are the same edge in your example above.

Reply:
Fixed.